site/app/controllers/forum/ForumController.php

```diff
@@ -339,6 +339,12 @@ public function publishThread(){
             return $this->core->getOutput()->getOutput();
         }
 
+        $markdown = false;
+
+        if(isset($_POST['markdown_status']) && ($_POST['markdown_status']=="1")){
```

Comment:
Assuming `$_POST['markdown_status']` can either be undefined, `"0"`, or `"1"`, we can replace these checks with just `$markdown = !empty($_POST['markdown_status']`.